pkg/service/service.go

```diff
@@ -1,13 +1,125 @@
 package service
 
 import (
+	"context"
+
 	runapi "github.com/GoogleCloudPlatform/cloud-run-release-operator/internal/run"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/pkg/config"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/api/run/v1"
 )
 
-// Client is the connection to update and obtain data about a single service.
+// Client is the connection to update and obtain data about a single service. It
+// contains a client to the Cloud Run API, information about the service and the
+// service object.
 type Client struct {
 	RunClient   runapi.Client
+	Service     *run.Service
 	Project     string
 	ServiceName string
 	Region      string
 }
+
+// NewClient initializes a service client.
+func NewClient(runclient runapi.Client, svc *run.Service, project, serviceName, region string) *Client {
+	return &Client{
+		RunClient:   runclient,
+		Service:     svc,
+		Project:     project,
+		ServiceName: serviceName,
+		Region:      region,
+	}
+}
+
+// Filter retrieves all the services that match the filter in the configuration.
+func Filter(logger *logrus.Logger, cfg *config.Config) []*Client {
```

Comment:
1. I think this method is poorly named. It sounds like QueryAndFilterAllServices()

2. You only use `cfg.Targets` so why are you passing `config.Config` here?